mypy/main.py

```diff
@@ -612,6 +612,11 @@ def add_invertible_flag(flag: str,
     # --local-partial-types disallows partial types spanning module top level and a function
     # (implicitly defined in fine-grained incremental mode)
     parser.add_argument('--local-partial-types', action='store_true', help=argparse.SUPPRESS)
+    # --tweaked-deps adds some more dependencies that are not semantically needed, but
+    # may be helpful to determine relative importance of classes and functions for overall
+    # type precision in a code base. It also _removes_ some deps, so this flag should be never
+    # used except for generating code stats.
+    parser.add_argument('--tweaked-deps', action='store_true', help=argparse.SUPPRESS)
```

Comment:
Maybe rename to something like `--logical-deps` or `--semantic-deps` and make this imply `--cache-fine-grained`? The motivation is that this makes the dependencies more independent of the details of how fine-grained incremental checking works.

Reply:
I went with `--logical-deps` it sounds better.